src/CodeViewer/ViewerTest.cpp

```diff
@@ -39,4 +41,172 @@ TEST(Viewer, DetermineLineNumberWidthInPixels) {
+  QTextDocument doc{};
+  doc.setPlainText("first line\nsecond line\nthird line");
+  ASSERT_EQ(doc.blockCount(), 3);
+
+  const LargestOccuringLineNumbers line_numbers =
+      SetAnnotatingContentInDocument(&doc, absl::Span<const AnnotatingLine>{});
+
+  ASSERT_TRUE(line_numbers.main_content.has_value());
+  ASSERT_EQ(line_numbers.main_content.value(), 3);
+
+  ASSERT_FALSE(line_numbers.annotating_lines.has_value());
+  EXPECT_EQ(doc.blockCount(), 3);
+
+  for (auto current_block = doc.begin(); current_block != doc.end();
+       current_block = current_block.next()) {
+    EXPECT_NE(current_block.userData(), nullptr);
+  }
+}
+
+TEST(Viewer, SetAnnotatingContentInDocumentFirst) {
+  QTextDocument doc{};
+  constexpr int kNumberOfLines = 3;
+  doc.setPlainText("first line\nsecond line\nthird line");
+  ASSERT_EQ(doc.blockCount(), kNumberOfLines);
+
+  std::array<AnnotatingLine, 1> lines{};
+  lines[0].reference_line = 1;
+  lines[0].line_contents = "first annotation";
+  lines[0].line_number = 42;
+
+  const LargestOccuringLineNumbers line_numbers = SetAnnotatingContentInDocument(&doc, lines);
+
+  ASSERT_TRUE(line_numbers.main_content.has_value());
+  ASSERT_EQ(line_numbers.main_content.value(), kNumberOfLines);
+
+  ASSERT_TRUE(line_numbers.annotating_lines.has_value());
+  ASSERT_EQ(line_numbers.annotating_lines.value(), 42);
+
+  for (auto current_block = doc.begin(); current_block != doc.end();
+       current_block = current_block.next()) {
+    EXPECT_NE(current_block.userData(), nullptr);
+  }
+
+  EXPECT_EQ(doc.begin().text().toStdString(), "first annotation");
+}
+
+TEST(Viewer, SetAnnotatingContentInDocumentConsecutive) {
+  QTextDocument doc{};
+  constexpr int kNumberOfLines = 3;
+  doc.setPlainText("first line\nsecond line\nthird line");
+  ASSERT_EQ(doc.blockCount(), kNumberOfLines);
+
+  std::array<AnnotatingLine, 2> lines{};
+  lines[0].reference_line = 1;
+  lines[0].line_contents = "first annotation";
+  lines[0].line_number = 42;
+  lines[1].reference_line = 2;
+  lines[1].line_contents = "second annotation";
+  lines[1].line_number = 43;
+
+  const LargestOccuringLineNumbers line_numbers = SetAnnotatingContentInDocument(&doc, lines);
+
+  ASSERT_TRUE(line_numbers.main_content.has_value());
+  ASSERT_EQ(line_numbers.main_content.value(), kNumberOfLines);
+
+  ASSERT_TRUE(line_numbers.annotating_lines.has_value());
+  ASSERT_EQ(line_numbers.annotating_lines.value(), 43);
+
+  EXPECT_EQ(doc.blockCount(), 5);
+
+  for (auto current_block = doc.begin(); current_block != doc.end();
+       current_block = current_block.next()) {
+    EXPECT_NE(current_block.userData(), nullptr);
+  }
+
+  EXPECT_EQ(doc.begin().text().toStdString(), "first annotation");
+  EXPECT_EQ(doc.begin().next().text().toStdString(), "first line");
+  EXPECT_EQ(doc.begin().next().next().text().toStdString(), "second annotation");
+  EXPECT_EQ(doc.begin().next().next().next().text().toStdString(), "second line");
```

Comment:
Checking the last line also doesn't hurt here.

Reply:
done.